Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs

```diff
@@ -1953,19 +2075,512 @@ rideTrackingView push state =
+          ] <> FontStyle.body2 TypoGraphy
+        , imageView
+            [ height $ V 12
+            , width  $ V 12
+            , gravity CENTER_VERTICAL
+            , accessibility DISABLE
+            , margin $ MarginRight 4
+            , imageWithFallback $ fetchImage FF_ASSET "ny_ic_info"
+            ]
+        , waitTimeView push state
+        ])]
+      else [])
+
+shineAnimation :: forall w . Int -> Int -> PrestoDOM (Effect Unit) w
+shineAnimation height' width' =
+  linearLayout
+  [ height $ MATCH_PARENT
+  , width $ WRAP_CONTENT
+  , gravity CENTER_VERTICAL
+  , cornerRadius 16.0
+  , clipChildren true
+  , clipToPadding true
+  ][ PrestoAnim.animationSet
+    [ PrestoAnim.Animation
+      [ PrestoAnim.duration 1500
+      , PrestoAnim.fromX $ (-100)
+      , PrestoAnim.toX $ (width')
+      , PrestoAnim.repeatCount PrestoAnim.Infinite
+      ] true
+    ] $ linearLayout
+        [ width $ V $ width'
+        , height $ MATCH_PARENT
+        , gravity CENTER_VERTICAL
+        ][ linearLayout
+          [ width $ V 10
+          , height $ V $ height'
+          , background Color.white200
+          , rotation 20.0
+          , cornerRadius 2.0
+          , margin $ MarginHorizontal 10 8
+          ][]
+         , linearLayout
+           [ width $ V 5
+           , height $ V $ height'
+           , background Color.white200
+           , rotation 20.0
+           , cornerRadius 2.0
+           , margin $ MarginRight 20
+           ][]
+        ]
+  ]
+
+waitTimeView :: forall w. (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM ( Effect Unit) w
+waitTimeView push state =
+  linearLayout
+  [ height WRAP_CONTENT
+  , width MATCH_PARENT
+  , cornerRadius if os == "IOS" then 12.0 else 16.0
+  , margin $ Margin 0 4 4 4
+  , background $ colorForWaitTime state
+  , gravity CENTER_VERTICAL
+  , accessibility DISABLE
+  ][ textView $ 
+     [ height $ WRAP_CONTENT 
+     , width $ V $ 60
+     , padding $ Padding 0 4 0 6
+     , accessibilityHint $ (if state.data.currentSearchResultType == QUOTES then "O T P Expires in : " else "Wait Time : ") <> case DS.split (DS.Pattern ":") state.data.driverInfoCardState.waitingTime of
+                                                [minutes, seconds] -> do 
+                                                  let min = DS.trim $ minutes
+                                                  let sec = DS.trim $ seconds
+                                                  if min /= "00" then min <> " Minutes and " <> sec <> " Seconds" else sec <> " Seconds" 
+                                                _ -> ""
+     , accessibility ENABLE 
+     , text $ state.data.driverInfoCardState.waitingTime
+     , color Color.black900
+     , gravity CENTER
+     , singleLine true
+     ] <> FontStyle.body4 TypoGraphy
+  ]
+
+colorForWaitTime:: HomeScreenState -> String
+colorForWaitTime state =
+  case DS.split (DS.Pattern ":") state.data.driverInfoCardState.waitingTime of
```

Comment:
`DS.split (DS.Pattern ":") state.data.driverInfoCardState.waitingTime` Put this in a variable and use here in case